tests/automation/setup/sessionTest.ts

```diff
@@ -72,27 +89,354 @@ export function sessionTestThreeWindows(
+            'linking a window with ',
+            users[link - 1].recoveryPhrase,
+          );
+          const linked = await linkedDevice(users[link - 1].recoveryPhrase);
+          linkedWindows.push(linked);
+        }
+      }
+
+      let groupCreated: Group | undefined;
+      if (grouped?.length) {
+        groupCreated = await createGroup(
+          testName,
+          users[grouped[0] - 1],
+          mainWindows[grouped[0] - 1],
+          users[grouped[1] - 1],
+          mainWindows[grouped[1] - 1],
+          users[grouped[2] - 1],
+          mainWindows[grouped[2] - 1],
+        );
+      }
+
+      // Sadly, we need to cast the parameters here, so our less generic function have correct types for the callback
+      await testCallback(
+        {
+          mainWindows: mainWindows as Tuple<Page, UserCount>,
+          linkedWindows: linkedWindows as Tuple<Page, Links['length']>,
+          users,
+          groupCreated: groupCreated as Grouped extends Array<any>
+            ? Group
+            : undefined,
+        },
+        testinfo,
+      );
+    } catch (e) {
+      throw e;
+    } finally {
+      try {
+        await forceCloseAllWindows([...mainWindows, ...linkedWindows]);
+      } catch (e) {
+        console.error(`forceCloseAllWindows of ${testName} failed with: `, e);
+      }
+    }
+  });
 }
 
-export function sessionTestFiveWindows(
-  testName: string,
-  testCallback: ([windowA, windowB, windowC, windowD]: [
-    Page,
-    Page,
-    Page,
-    Page,
-    Page,
-  ]) => Promise<void>,
+/**
+ * Setup the test with 1 user and a single window, but don't wait for the network to be ready.
+ * Used for tests which don't need network (i.e. setting/checking passwords etc)
+ */
+export function test_Alice_1W_no_network(
+  testname: string,
+  testCallback: (
+    details: WithAlice & WithaliceWindow1,
+    testInfo: TestInfo,
+  ) => Promise<void>,
+) {
+  return sessionTestGeneric(
+    testname,
+    1,
+    { waitForNetwork: false },
+    ({ mainWindows, users }, testInfo) => {
+      return testCallback(
+        {
+          alice: users[0],
+          aliceWindow1: mainWindows[0],
+        },
+        testInfo,
+      );
+    },
+  );
+}
+
+/**
+ * Setup the test with 1 user and 2 windows total:
+ * - Alice with 2 windows.
+ */
+export function test_Alice_2W(
+  testname: string,
+  testCallback: (
+    details: WithAlice & WithaliceWindow1 & WithaliceWindow2,
+    testInfo: TestInfo,
+  ) => Promise<void>,
+) {
+  return sessionTestGeneric(
+    testname,
+    1,
+    { links: [1] },
+    ({ mainWindows, users, linkedWindows }, testInfo) => {
+      return testCallback(
+        {
+          alice: users[0],
+          aliceWindow1: mainWindows[0],
+          aliceWindow2: linkedWindows[0],
+        },
+        testInfo,
+      );
+    },
+  );
+}
+
+/**
+ * Setup the test with 2 users and 2 windows total:
+ * - Alice with 1 window,
+ * - Bob with 1 window.
+ */
+export function test_Alice_1W_Bob_1W(
```

Comment:
Are fixtures snake case by convention? Or just for readability?

Reply:
just for readability.
testAlice1WBob1W, is not awesome to read

Reply:
OK